src/modules/posts/usecase/usecase.ts

```diff
@@ -65,6 +65,21 @@ class Usecase {
 
         return data
     }
+
+    public async Delete(id: string, database: string) {
+        const item = await this.repository.FindById(id, database)
+
+        if (!item)
+            throw new error(
+                statusCode.NOT_FOUND,
+                statusCode[statusCode.NOT_FOUND]
+            )
+
+        const result = await this.repository.Delete(id, database)
+        await this.repository.DecreaseCategory(item.category, database)
```

Comment:
```suggestion
        await this.repository.DecreaseCategory(item.category.id, database)
```
sorry for `item.category` what is the value?

Reply:
it is `string` of the `id` category @ayocodingit